src/main/java/com/cburch/logisim/fpga/data/IOComponentTypes.java

```diff
@@ -288,32 +436,111 @@ public static void paintPartialMap(Graphics2D g,
+          for (var yIndex = 0; yIndex < 7; yIndex++) {
+            if (indexes[yIndex][xIndex] == pinNr) {
+              switch (mapRotation) {
+                case rotationPlusNinety : {
+                  realXIndex = yIndex;
+                  realXIndexPlusOne = yIndex  + 1;
+                  realYIndex = 4 - xIndex;
+                  realYIndexPlusOne = 5 - xIndex;
+                  break;
+                }
+                case rotationMinusNinety : {
+                  realXIndex = 6 - yIndex;
+                  realXIndexPlusOne = 7 - yIndex;
+                  realYIndex = xIndex;
+                  realYIndexPlusOne = xIndex + 1;
+                  break;
+                }
+                default : {
+                  realXIndex = xIndex;
+                  realXIndexPlusOne = xIndex + 1;
+                  realYIndex = yIndex;
+                  realYIndexPlusOne = yIndex + 1;
+                  break;
+                }
+              }
+              final var boxX = x + (int) ((float) realXIndex * partX);
+              final var boxY = y + (int) ((float) realYIndex * partY);
+              /* the below calculation we do to avoid truncation errors causing empty lines between the segments */
+              final var boxWidth = (int) ((float) realXIndexPlusOne * partX) - (int) ((float) realXIndex * partX);
+              final var boxHeight = (int) ((float) realYIndexPlusOne * partY) - (int) ((float) realYIndex * partY);
+              g.fillRect(boxX, boxY, boxWidth, boxHeight);
             }
           }
         }
         break;
       }
       case LEDArray: {
-        final var partx = (float) width / (float) nrOfColumns;
-        final var party = (float) height / (float) nrOfRows;
-        final var xPos = pinNr % nrOfColumns;
-        final var yPos = pinNr / nrOfColumns;
-        final var bx = x + (int) ((float) xPos * partx);
-        final var by = y + (int) ((float) yPos * party);
-        final var bw = x + (int) ((float) (xPos + 1) * partx) - bx;
-        final var bh = y + (int) ((float) (yPos + 1) * party) - by;
+        var partX = (float) 0;
+        var partY = (float) 0;
+        final var selectedColumn = pinNr % nrOfColumns;
+        final var selectedRow = pinNr / nrOfColumns;
+        switch (mapRotation) {
+          case rotationPlusNinety :
+          case rotationMinusNinety : {
+            partX = (float) width / (float) nrOfRows;
+            partY = (float) height / (float) nrOfColumns;
+            break;
+          }
+          default : {
+            partX = (float) width / (float) nrOfColumns;
+            partY = (float) height / (float) nrOfRows;
+            break;
+          }
+        }
+        var xPosition = 0;
+        var nextXPosition = 0;
+        var yPosition = 0;
+        var nextYPosition = 0;
+        switch (mapRotation) {
+          case rotationPlusNinety : {
+            xPosition = (int) ((float) selectedRow * partX);
```

Comment:
Is explicit cast to `float` in `(int)((float)x * y)` type-o-lines really needed?